barretenberg/cpp/src/barretenberg/sumcheck/sumcheck_round.hpp

```diff
@@ -526,15 +532,18 @@ template <typename Flavor> class SumcheckVerifierRound {
                                              const RelationSeparator alpha,
                                              std::optional<FF> full_libra_purported_value = std::nullopt)
     {
-        // The verifier should never skip computation of contributions from any relation
         Utils::template accumulate_relation_evaluations_without_skipping<>(
             purported_evaluations, relation_evaluations, relation_parameters, gate_sparators.partial_evaluation_result);
 
         FF running_challenge{ 1 };
         FF output{ 0 };
         Utils::scale_and_batch_elements(relation_evaluations, alpha, running_challenge, output);
+
         if constexpr (Flavor::HasZK) {
             output += full_libra_purported_value.value();
+            if constexpr (IsECCVMRecursiveFlavor<Flavor>) {
+                output.self_reduce();
```

Comment:
we have to self_reduce here

Reply:
this is only necessary for the ECCVM? is it clear why?